### ISSUE TITLE
drupal playbook: /etc/php5/apache2/conf.d/20-apcu.ini does not exist

### ISSUE THREAD

**Issue by e1nh4nd3r** [geerlingguy/ansible-for-devops#34]
I encountered the below problem and was able to reproduce it and verify that in fact the file is not present.

```
...
TASK [Install Apache, MySQL, PHP, and other dependencies.] *********************
changed: [default] => (item=[u'git', u'curl', u'sendmail', u'apache2', u'php5', u'php5-common', u'php5-mysql', u'php5-cli', u'php5-curl', u'php5-gd', u'php5-dev', u'php5-mcrypt', u'php-apc', u'php-pear', u'python-mysqldb', u'mysql-server'])

TASK [Disable the firewall (since this is for local dev only).] ****************
changed: [default]

TASK [Start Apache, MySQL, and PHP.] *******************************************
ok: [default] => (item=apache2)
ok: [default] => (item=mysql)

TASK [Enable Apache rewrite module (required for Drupal).] *********************
changed: [default]

TASK [Add Apache virtualhost for Drupal 8 development.] ************************
changed: [default]

TASK [Symlink Drupal virtualhost to sites-enabled.] ****************************
changed: [default]

TASK [Remove default virtualhost file.] ****************************************
changed: [default]

TASK [Enable upload progress via APC.] *****************************************
fatal: [default]: FAILED! => {"changed": false, "failed": true, "msg": "Destination /etc/php5/apache2/conf.d/20-apcu.ini does not exist !", "rc": 257}

RUNNING HANDLER [restart apache] ***********************************************

PLAY RECAP *********************************************************************
default                    : ok=11   changed=7    unreachable=0    failed=1

Ansible failed to complete successfully. Any error output should be
visible above. Please fix these errors and try again.
```

Using `find` from `/`:
```
root@vagrant:/# find . -name '*apcu*'
./etc/php5/cli/conf.d/20-apcu.ini
./etc/php5/cgi/conf.d/20-apcu.ini
./etc/php5/mods-available/apcu.ini
./usr/lib/php5/20121212/apcu.so
./usr/include/php5/ext/apcu
./usr/share/doc/php5-apcu
./usr/share/php/.registry/.channel.pecl.php.net/apcu.reg
./var/lib/php5/modules/cli/enabled_by_maint/apcu
./var/lib/php5/modules/cgi/enabled_by_maint/apcu
./var/lib/php5/modules/registry/apcu
./var/lib/dpkg/info/php5-apcu.md5sums
./var/lib/dpkg/info/php5-apcu.postrm
./var/lib/dpkg/info/php5-apcu.conffiles
./var/lib/dpkg/info/php5-apcu.prerm
./var/lib/dpkg/info/php5-apcu.list
./var/lib/dpkg/info/php5-apcu.postinst
./var/cache/apt/archives/php5-apcu_4.0.10-1+deb.sury.org~precise+1_amd64.deb
root@vagrant:/#
```

Unsure which of the files this is supposed to be actioned against.  I'm assuming both `cli` and `cgi` since I'd imagine one dir is for CLI invocation of `php` and the other is for `apache`?

**Comment by e1nh4nd3r** [comment]
Changing the below line seems to have allowed the playbook to execute successfully.
```
~/ansible-for-devops/drupal (master)$ git diff
diff --git a/drupal/provisioning/playbook.yml b/drupal/provisioning/playbook.yml
index 3167593..520dc9d 100644
--- a/drupal/provisioning/playbook.yml
+++ b/drupal/provisioning/playbook.yml
@@ -79,7 +79,7 @@

     - name: Enable upload progress via APC.
       lineinfile:
-        dest: "/etc/php5/apache2/conf.d/20-apcu.ini"
+        dest: "/etc/php5/cgi/conf.d/20-apcu.ini"
         regexp: "^apc.rfc1867"
         line: "apc.rfc1867 = 1"
         state: present
```

Unsure if that's the correct way to fix this or if editing both files under `cgi` and `cli` is better.

**Comment by e1nh4nd3r** [comment]
Going to test with that entire block commented.  Distinctly possible that in the latest packaged version of php5 that is installed that the option is enabled by default.  Stand-by.

**Comment by e1nh4nd3r** [comment]
Helps if I look at the actual file.  Looks like it's now a symlink to something else.
```
root@vagrant:/etc# ls -lah ./php5/cli/conf.d/20-apcu.ini
lrwxrwxrwx 1 root root 29 May 16 21:45 ./php5/cli/conf.d/20-apcu.ini -> ../../mods-available/apcu.ini
```

Editing to match the path and re-running.

**Comment by e1nh4nd3r** [comment]
That worked.  Here's the diff:
```
~/ansible-for-devops/drupal (master)$ git diff
diff --git a/drupal/provisioning/playbook.yml b/drupal/provisioning/playbook.yml
index 3167593..0895729 100644
--- a/drupal/provisioning/playbook.yml
+++ b/drupal/provisioning/playbook.yml
@@ -79,7 +79,7 @@

     - name: Enable upload progress via APC.
       lineinfile:
-        dest: "/etc/php5/apache2/conf.d/20-apcu.ini"
+        dest: "/etc/php5/mods-available/apcu.ini"
         regexp: "^apc.rfc1867"
         line: "apc.rfc1867 = 1"
         state: present
```

I'll fork and submit a PR.

**Comment by e1nh4nd3r** [comment]
Created a PR for this fix.
https://github.com/geerlingguy/ansible-for-devops/pull/35

**Comment by szop85** [comment]
I guess it's a different problem. Since this playbook is proving PHP via PPA ondrej/php5, there is no compatible package for libapache2-mod-php5, which would create /etc/php5/apache2. ondrej/php5 is expecting Apache 2.4 not 2.2 which is installed by default in Ubuntu 12.04.

**Comment by geerlingguy** [comment]
I'm taking a quick look at this; part of the fix will be updating the book example to 14.04 (as well as here, in the Vagrantfile).

**Comment by geerlingguy** [comment]
This should be working with the latest changes (see the above commit https://github.com/geerlingguy/ansible-for-devops/commit/cbf1974ee0c84de29c5eca6dfb5d3c05515fe029).
